django_s3/tests/test_tus_upload.py

```diff
@@ -392,3 +392,174 @@ def test_post_returns_500_on_create_initial_file_error(self):
                         mock.Mock(side_effect=Exception("fail"))):
             resp = view.post(request)
             self.assertEqual(resp.status_code, 500)
+
+    def test_post_returns_413_when_user_exceeds_quota(self):
+        """Test that POST returns 413 when user exceeds quota"""
+        view = CustomTusUpload()
+        request = self.factory.post("/")
+        request.user = self.user
+
+        # Set HTTP_UPLOAD_LENGTH to a large value to exceed quota
+        self.file_size = 1000000000000
+
+        self.create_request_metadata(request)
+        view.request = request
+        view.kwargs = {'resource_id': self.res.short_id}
+
+        resp = view.post(request)
+        # Should return 413 if quota is exceeded
+        self.assertEqual(resp.status_code, 413)
+
+    def test_post_uses_metadata_file_size_when_no_upload_length(self):
+        """Test that POST uses file_size from metadata when HTTP_UPLOAD_LENGTH is missing"""
+        view = CustomTusUpload()
+        request = self.factory.post("/")
+        request.user = self.user
+        request.META["HTTP_TUS_RESUMABLE"] = "1.0.0"
+        # Don't set HTTP_UPLOAD_LENGTH - should use metadata file_size
+        self.create_request_metadata(request)
+        view.request = request
+        view.kwargs = {'resource_id': self.res.short_id}
+
+        resp = view.post(request)
+        # Should succeed and use file_size from metadata
+        self.assertEqual(resp.status_code, 201)
+
+    def test_post_prioritizes_upload_length_over_metadata_file_size(self):
+        """Test that HTTP_UPLOAD_LENGTH takes precedence over metadata file_size"""
+        view = CustomTusUpload()
+        request = self.factory.post("/")
+        request.user = self.user
+        request.META["HTTP_TUS_RESUMABLE"] = "1.0.0"
+        request.META["HTTP_UPLOAD_LENGTH"] = "500"  # Different from metadata file_size
+        self.create_request_metadata(request)  # Contains file_size = self.file_size
+        view.request = request
+        view.kwargs = {'resource_id': self.res.short_id}
+
+        resp = view.post(request)
+        # Should succeed using HTTP_UPLOAD_LENGTH (500) instead of metadata file_size
+        self.assertEqual(resp.status_code, 201)
+
+    def test_post_handles_null_metadata_file_size(self):
+        """Test that POST handles 'null' string in metadata file_size"""
+        view = CustomTusUpload()
+        request = self.factory.post("/")
+        request.user = self.user
+        request.META["HTTP_TUS_RESUMABLE"] = "1.0.0"
+        request.META["HTTP_UPLOAD_LENGTH"] = "300"
+
+        # Modify the metadata to have 'null' file_size
+        metadata = self.fake_metadata()
+        metadata['file_size'] = 'null'
```

Comment:
Thank you fro adding these tests. I'm wondering if we could simplify file_size handling if we updated file_size to be 0 when a [Value/Type Error is caught](https://github.com/hydroshare/hydroshare/pull/6044/files#diff-710124ee850cc9123e4b5ac53a3a3cbb4bdd77ccd7fecbfaa707a63aa2e47e99R397) instead of the 'null' string.

Reply:
did that, thanks